README.md

```diff
@@ -127,8 +127,50 @@ bootstrap(App, [
 
 ### FirebaseAuth
 
-Injectable service for managing authentication state. It currently only supports
-reading auth state.
+Injectable service for managing authentication state.
+
+#### Logging In
+To log in a user, call the `login` method on an instance of `FirebaseAuth` class. The method has
+the following two signatures:
+
+```typescript
+login(config?: AuthConfiguration): Promise<FirebaseAuthState>;
+login(credentials: AuthCredentials, config?: AuthConfiguration): Promise<FirebaseAuthState>;
+```
+The signature that is used depends on which AuthMethod you chose to use to login.
+AuthMethods.Popup, AuthMethods.Redirect, and AuthMethods.Anonymous all use the first signature whereas
+AuthMethods.CustomToken, AuthMethods.OAuthToken, and AuthMethods.Password use the second signature. This is
+because if you use these three AuthMethods you need to provide a credentials argument to login.
+
+##### AuthConfiguration
+You **MUST** provide an `AuthConfiguration` object to use the `login` method, however you do not need
+to pass it to login correctly. Instead you may choose to pass the configuration in through DI. This helps
+keep your components modular because they can simply call `login` and it will use whichever options were
+provided through DI.
+You can use the `
+```typescript
+bootstrap(app, [
+  FIREBASE_PROVIDERS,
+  defaultFirebase('https://<some-firebase>.firebaseio.com'),
+  firebaseAuthConfig({
```

Comment:
It would be good to include the `import` statements here (and in our other examples going forward) to be clear what is coming from where.